Add-Kedro viz-Main-Git

rxm7706 opened the pull request:
Goal - Create a conda package directly from the python src distribution
      0 https://pypi.org/project/kedro-viz/#files

Current state - 8/12/2023
 - Only able to build by pointing to git repo directly (main)

Known Issues - to get to building off pypi packaged source
 - Licenses are not packaged
 - package.json is not included. 
 - Unsure if nodejs 18 is supported, upstream is using nodejs 16
 - possible issue with path in medata-conda-themes
    -    '../../../../node_modules/highlight.js/styles/a11y-light'
    -    '../../../../node_modules/highlight.js/styles/a11y-dark'
    - https://github.com/kedro-org/kedro-viz/blob/6e1d127fb85349ca7244e76719b756cc5001784f/src/components/metadata/styles/metadata-code-themes.scss#L5
 - possible issue requiring   
    -  NODE_OPTIONS=--openssl-legacy-provider npm run build
    -  Unable to build locally 

<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Gitter channel][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io
-->

Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

rxm7706 commented:
@conda-forge-admin, please rerender

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/kedro-viz```) and found it was in an excellent condition.

rxm7706 commented:
@conda-forge-admin, please rerender

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/kedro-viz```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipes/kedro-viz**:

 * License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression.

Documentation on acceptable licenses can be found [here]( https://conda-forge.org/docs/maintainer/adding_pkgs.html#spdx-identifiers-and-expressions ).

rxm7706 commented:
@conda-forge-admin, please rerender

astrojuanlu commented:
Hi @rxm7706! FYI I tried to give this a stab as well, cherry picking your branch and poking around https://github.com/conda-forge/staged-recipes/pull/22142 still haven't had time to finish it, but at least reported a couple of issues upstream that you might want to look into: https://github.com/kedro-org/kedro-viz/issues/1267 (hence sdists cannot be used) and https://github.com/kedro-org/kedro-viz/issues/1278 (hence requiring `NODE_OPTIONS=--openssl-legacy-provider`).

rxm7706 commented:
> Hi @rxm7706! FYI I tried to give this a stab as well, cherry picking your branch and poking around #22142 still haven't had time to finish it, but at least reported a couple of issues upstream that you might want to look into: [kedro-org/kedro-viz#1267](https://github.com/kedro-org/kedro-viz/issues/1267) (hence sdists cannot be used) and [kedro-org/kedro-viz#1278](https://github.com/kedro-org/kedro-viz/issues/1278) (hence requiring `NODE_OPTIONS=--openssl-legacy-provider`).

Thanks @astrojuanlu ; taking a look - I've added your NODE_OPTIONS change, updated the pypi version and package.json, also added you to maintainers. This should at least prevent mixing issues related to the older version on pypi and outdated package.json

rxm7706 commented:
@conda-forge-admin, please rerender

rxm7706 commented:
@conda-forge-admin, please rerender

astrojuanlu commented:
😮 so the build process is working, finally?

astrojuanlu commented on recipes/kedro-viz/package.json:
```diff
@@ -0,0 +1,166 @@
+{
```
This would force us to keep this `package.json` and the one upstream in sync, can't we get it from the repository?

rxm7706 commented:
Quite a bit still needs to be done upstream - since we need to have the build working off the packaged src from pypi.. (not github src) - yes, package.json will need to be included into the pypi src package (also the license files)

rxm7706 commented:
@conda-forge-admin, please rerender

rxm7706 commented:
@astrojuanlu I've summarized all issues I am aware of in the title. Please take a look, and let me know if you have any additional comments / suggestions on how to get this moving again

astrojuanlu commented:
> how to get this moving again

I think we need to work upstream on https://github.com/kedro-org/kedro-viz/.

See https://github.com/kedro-org/kedro-viz/issues/1267#issuecomment-1565240967 for some ideas on how to solve the JS part (the difficult one). For the licenses, it should be as easy as modifying `MANIFEST.in` accordingly.

If we bundle the JS files with the sdist upstream, then the conda-forge recipe would not need Node and we could skip the Node 18 problem, if I'm not mistaken.

rxm7706 commented:
closing this - as https://anaconda.org/conda-forge/kedro-viz is now available and  the feedstock is working as expected.

To summarize 
1. The gap seemed to come down to differences between the targ.gz on pypi vs using the source tar ball form the kedro github repo..  
2. Also changing pip install to pick the package directory. 
3. This only fixes conda install, and some of the open upstream issues with pip install / making source tarball on pip and github release match - still exist.

astrojuanlu commented:
Oh wow, I was unaware of #23679, amazing! Didn't occur to me to use the tarball from GH. Still not 100 % sure why it works, but glad it does anyway. Thanks @rxm7706 for your effort!

astrojuanlu commented:
Sadly it was not as easy, issue incoming.
